src/renderer/sagas/socket/socket.saga.ts

```diff
@@ -107,12 +126,117 @@ export function* getPublicChannels(socket): Generator {
   }
 }
 
+// Direct Messages
+
+export function* subscribeForDirectMessageThread(socket): Generator {
+  while (true) {
+    const { payload } = yield* take(`${directMessagesActions.subscribeForDirectMessageThread}`)
+    socket.emit(socketsActions.SUBSCRIBE_FOR_DIRECT_MESSAGE_THREAD, payload)
+  }
+}
+
+export function* getAvailableUsers(socket): Generator {
+  while (true) {
```

Comment:
this `while (true)` is kind of not nice. I know you've been following what Janusz did, but apparently I haven't reviewed his code enough.

Please change this as follows:

1. In `useIO` below, instead of `fork` do:

takeEvery(directMessagesActions.getAvailableUsers.type, getAvailableUsers, socket)

2. Change `getAvailableUsers` to:

```
export function* getAvailableUsers(socket): Generator {
  yield * call (socket.emit, socketsActions.GET_AVAILABLE_USERS)
}
```
Less code right?

Pleae do this in entire file for all the effects